Project1/MapHelper.h

```diff
@@ -4,32 +4,51 @@
 #include "Unit.h"
 #include <vector>
 
-class GridMap;
+#include "GridMap.h"
+
 class Range;
 
 class MapHelper
 {
 protected:
 	GridMap& _map;
 	MapHelper(GridMap& map);
-	~MapHelper() = default;
 public:
-	
-	std::vector<UnitRef> getUnits();
-	std::vector<UnitRef> getAllies(const Unit& unit);
-	std::vector<UnitRef> getEnemies(const Unit& unit);
-
-	std::vector<UnitRef> getUnitsWithin(const Unit& unit, Range range);
-	std::vector<UnitRef> getAlliesWithin(const Unit& unit, Range range);
-	std::vector<UnitRef> getEnemiesWithin(const Unit& unit, Range range);
-
-	std::vector<UnitRef> getUnitsWithin(Range range, GridCell& pos);
-	std::vector<UnitRef> getAlliesWithin(Range range, GridCell& pos, const PartyBase& party);
-	std::vector<UnitRef> getEnemiesWithin(Range range, GridCell& pos, const PartyBase& party);
-
-	std::vector<CellRef> getCellsWithin(Range range, GridCell & pos);
-	std::vector<CellRef> getAlliedCellsWithin(Range range, GridCell & pos, const PartyBase& party);
-	std::vector<CellRef> getEnemyCellsWithin(Range range, GridCell & pos, const PartyBase& party);
-	//std::vector<CellRef> getCellsWithin(Range range, GridCell& pos);
-};
+	~MapHelper() = default;
+
+	std::vector<Unit::Ref> getUnits();
+	std::vector<Unit::Ref> getAllies(const Unit& unit);
+	std::vector<Unit::Ref> getEnemies(const Unit& unit);
+
+	std::vector<Unit::Ref> getUnitsWithin(const Unit& unit, Range range);
+	std::vector<Unit::Ref> getAlliesWithin(const Unit& unit, Range range);
+	std::vector<Unit::Ref> getEnemiesWithin(const Unit& unit, Range range);
+
+	std::vector<Unit::Ref> getUnitsWithin(Range range, ID pos);
+	std::vector<Unit::Ref> getAlliesWithin(Range range, ID pos, const PartyBase& party);
+	std::vector<Unit::Ref> getEnemiesWithin(Range range, ID pos, const PartyBase& party);
 
+	std::vector<GridCell::Ref> getCellsWithin(Range range, ID pos);
+	std::vector<GridCell::Ref> getAlliedCellsWithin(Range range, ID pos, const PartyBase& party);
+	std::vector<GridCell::Ref> getEnemyCellsWithin(Range range, ID pos, const PartyBase& party);
+
+	const std::vector<Unit::Ref> getUnits() const;
```

Comment:
same thing as above with the const vectors; I don't think this does what you want.